src/math/matrix.c

```diff
@@ -119,28 +119,27 @@ int mat_multiply_elementwise(struct mat_matrix_16b *a, struct mat_matrix_16b *b,
 	int16_t *x = a->data;
 	int16_t *y = b->data;
 	int16_t *z = c->data;
-	int64_t p;
+	int32_t prod;
 	int i;
-	const int shift_minus_one = a->fractions + b->fractions - c->fractions - 1;
```

Comment:
same in this commit - please check that the commit description still matches the contents and I'd like to be able to see that too. As it stands, it's difficult to see where "unnecessary conditionals" are eliminated